src/buildercore/cfngen.py

```diff
@@ -350,34 +372,46 @@ def generate_stack(pname, **more_context):
     """given a project name and any context overrides, generates a Cloudformation
     stack file, writes it to file and returns a pair of (context, stackfilename)"""
     context = build_context(pname, **more_context)
-    template = render_template(context)
+    cloudformation_template = render_template(context)
+    terraform_template = terraform.render(context)
     stackname = context['stackname']
-    out_fname = write_template(stackname, template)
+
     context_handler.write_context(stackname, context)
-    return context, out_fname
+    cloudformation_template_file = write_template(stackname, cloudformation_template)
+    terraform_template_file = write_terraform_template(stackname, terraform_template)
+    return context, cloudformation_template_file, terraform_template_file
```

Comment:
should become `context, data_structure` where `data_structure` could be a list of two elements or any way to keep these files together. `namedtuple` maybe

Reply:
eh, namedtuples have their place but it isn't here, use a plain old dict

Reply:
... do we see ourselves writing a third provider? I think if anything I'd rather write a provider for terraform and access it that way. in that case, I'd just leave it as a triple,